File: client/src/lib/socketEvents.js
/**
 * BlancBleu — Miroir client des noms d'events Socket.IO.
 *
 * SYNCHRONISÉ À LA MAIN avec server/sockets/events.js (source de vérité).
 * Si tu touches ici, touche aussi le fichier serveur (et inversement).
 *
 * Voir docs/socket-events.md pour le tableau complet {event, room, payload}.
 */

const SOCKET_EVENTS = Object.freeze({
  // ── Serveur → clients (room transport:{id}) ─────────────────────────────────
  TRANSPORT_STATUS:    "transport:status",
  TRANSPORT_GPS:       "transport:gps",
  TRANSPORT_ASSIGNED:  "transport:assigned",
  TRANSPORT_CANCELLED: "transport:cancelled",
  TRANSPORT_CREATED:   "transport:created",
  TRANSPORT_UPDATED:   "transport:updated",
  TRANSPORT_SIGNATURE: "transport:signature_added",

  // ── Serveur → staff (rooms role:dispatcher / role:admin / role:superviseur) ─
  VEHICLE_POSITION:    "vehicle:position",
  VEHICLE_SNAPSHOT:    "vehicle:positions_snapshot",
  VEHICLE_STATUS:      "vehicle:status_changed",
  DRIVER_ONLINE:       "driver:online",
  DRIVER_OFFLINE:      "driver:offline",
  DRIVER_STATUS:       "driver:status_changed",
  DISPATCHER_STATUS:   "dispatcher:status",
  STATS_UPDATE:        "stats:update",
  PMT_EXTRACTED:       "pmt:extraite",

  // ── Auto-dispatch ───────────────────────────────────────────────────────────
  AUTODISPATCH_PROPOSAL: "autoDispatch:proposal_created",
  AUTODISPATCH_ASSIGNED: "autoDispatch:auto_assigned",
  AUTODISPATCH_DECIDED:  "autoDispatch:proposal_decided",

  // ── Messagerie ──────────────────────────────────────────────────────────────
  MESSAGE_TO_DRIVER:     "message:dispatcher",
  MESSAGE_TO_DISPATCHER: "message:driver",
  MESSAGE_DELIVERED:     "message:delivered",
  SHIFT_FORCED_END:      "shift:forced_end",
  NOTIFICATION_UNREAD:   "notification:unread_count",

  // ── Système ─────────────────────────────────────────────────────────────────
  SYSTEM_HEARTBEAT:    "system:heartbeat",
  CONNECTED_ACK:       "connected:ack",

  // ── Client → serveur (entrants) ─────────────────────────────────────────────
  IN_DRIVER_LOCATION:    "driver:location",
  IN_DRIVER_STATUS:      "driver:status",
  IN_MESSAGE_DRIVER:     "message:driver",
  IN_MESSAGE_DISPATCHER: "message:dispatcher",
  IN_JOIN_TRANSPORT:     "join:transport",
  IN_LEAVE_TRANSPORT:    "leave:transport",
  IN_JOIN_ROLE:          "join:role",
  IN_SHIFT_FORCE_END:    "shift:force_end",
});

export default SOCKET_EVENTS;
export { SOCKET_EVENTS };

File: packages/bb_core/lib/src/events/socket_events.dart
/// Miroir Dart des noms d'events Socket.IO.
///
/// SYNCHRONISÉ À LA MAIN avec `server/sockets/events.js` (source de vérité).
/// Si tu touches ici, touche aussi le fichier serveur.
///
/// Voir `docs/socket-events.md` pour le tableau complet {event, room, payload}.
///
/// Sprint M3 : consolidation des miroirs auparavant dupliqués dans chaque app
/// (`blancbleu_driver/lib/core/network/socket_events.dart` et idem patient).
/// Les apps ré-exportent désormais depuis ce package.
library;

class SocketEvents {
  SocketEvents._();

  // ── Serveur → clients (room transport:{id}) ─────────────────────────────
  static const String transportStatus    = "transport:status";
  static const String transportGps       = "transport:gps";
  static const String transportAssigned  = "transport:assigned";
  static const String transportCancelled = "transport:cancelled";
  static const String transportCreated   = "transport:created";
  static const String transportUpdated   = "transport:updated";
  static const String transportSignature = "transport:signature_added";

  // ── Serveur → staff (rooms role:dispatcher / role:admin / role:superviseur)
  static const String vehiclePosition    = "vehicle:position";
  static const String vehicleSnapshot    = "vehicle:positions_snapshot";
  static const String vehicleStatus      = "vehicle:status_changed";
  static const String driverOnline       = "driver:online";
  static const String driverOffline      = "driver:offline";
  static const String driverStatus       = "driver:status_changed";
  static const String dispatcherStatus   = "dispatcher:status";
  static const String statsUpdate        = "stats:update";
  static const String pmtExtracted       = "pmt:extraite";

  // ── Auto-dispatch (Sprint 6) ────────────────────────────────────────────
  static const String autoDispatchProposal = "autoDispatch:proposal_created";
  static const String autoDispatchAssigned = "autoDispatch:auto_assigned";
  static const String autoDispatchDecided  = "autoDispatch:proposal_decided";

  // ── Messagerie ──────────────────────────────────────────────────────────
  static const String messageDispatcher  = "message:dispatcher"; // serveur → chauffeur
  static const String messageDriver      = "message:driver";     // serveur → dispatcher
  static const String messageDelivered   = "message:delivered";
  static const String shiftForcedEnd     = "shift:forced_end";
  static const String notificationUnread = "notification:unread_count";

  // ── Système ────────────────────────────────────────────────────────────
  static const String systemHeartbeat    = "system:heartbeat";
  static const String connectedAck       = "connected:ack";

  // ── Client → serveur (entrants) ────────────────────────────────────────
  // NB : `message:driver` et `message:dispatcher` apparaissent dans les deux
  // sens — le nom sur le wire est identique. Côté serveur on les préfixe IN_
  // pour distinguer ; côté client on n'a pas d'ambiguïté car on est forcément
  // l'émetteur quand on appelle `socket.emit(...)`.
  static const String inDriverLocation   = "driver:location";
  static const String inDriverStatus     = "driver:status";
  static const String inMessageDriver    = "message:driver";
  static const String inMessageDispatcher = "message:dispatcher";
  static const String inJoinTransport    = "join:transport";
  static const String inLeaveTransport   = "leave:transport";
  static const String inJoinRole         = "join:role";
  static const String inShiftForceEnd    = "shift:force_end";
  static const String inPatientFcmToken  = "patient:fcm_token";
  static const String inRequestStats     = "request:stats";
}
